R/Steph_Function_Attempts.R

```diff
@@ -0,0 +1,101 @@
+
+#dummy package
+pkg <- arcticdatautils::create_dummy_package(mnTest,
+                                             size = 2)
+#dummy data table:
+attributes1 <- data.frame(
+    attributeName = c('col1', 'col2'),
+    attributeDefinition = c('Numbers', 'Letters in the alphabet'),
+    measurementScale = c('ratio', 'nominal'),
+    domain = c('numericDomain', 'textDomain'),
+    formatString = c(NA,NA),
+    definition = c(NA,'ABCDEFG...'),
+    unit = c('dimensionless', NA),
+    numberType = c('integer', NA),
+    missingValueCode = c(NA, NA),
+    missingValueCodeExplanation = c(NA, NA),
+    stringsAsFactors = FALSE)
+
+attributeList1 <- set_attributes(attributes1)
+phys <- pid_to_eml_physical(mnTest, pkg$data[1])
+
+dummy_data_table <- new('dataTable',
+                        entityName = 'Dummy Data Table',
+                        entityDescription = 'Dummy Description',
+                        physical = phys,
+                        attributeList = attributeList1)
+
+eml <- read_eml(rawToChar(getObject(mnTest, pkg$metadata)))
+eml@dataset@dataTable <- c(dummy_data_table)
+
+pkg #getting PIDs
+# $metadata
+# [1] "urn:uuid:83965425-2125-4663-bac7-d471fb03fb43"
+# $resource_map
+# [1] "urn:uuid:3a26404a-a8d0-4a6b-b081-3d6d96b40923"
+# $data
+# [1] "urn:uuid:9f1a87e5-0c1a-4498-b787-afe40d417be2"
+
+eml@dataset@dataTable@.Data
+#Need some way to access the <attributeList> inside this slot: maybe write in the xml text?
+
+#arguments would be listed like so, where x is the attribute of interest
+#AttName<-edit_attribute(x, attributeName="",attributeLabel="",attributeDefinition="",measurementScale="",
+                            #domain="",formatString="",definition="",unit="",numberType="",
+                            #missingValueCode="",missingValueCodeExplanation="")
+
+##Don't use this
+# edit_attribute<-function(attributeName="",attributeDefinition="",measurementScale="",
+#                          domain="",formatString="",definition="",unit="",numberType="",
+#                          missingValueCode="",missingValueCodeExplanation=""){
+#     paste0('<attribute>
+#              <attributeName>',attributeName,'</attributeName>',
+#              '<attributeDefinition>',attributeDefinition,'</attributeDefinition>',
+#              '<measurementScale><',measurementScale,'>',
+#                 '<unit><standardUnit>',unit,'</standardUnit></unit>',
+#                 '<',domain,'>','<numberType>',numberType,'</numberType></',domain,'>',
+#              '</measurementScale></attribute>')
+# }
+
+#Code below includes only the required slots and does not contain any if statements for failing the function if e.g. the wrong domain goes into a measurement scale.
+#x is the data table number in the data table list, y is the attribute number in the attribute list for that data table
+edit_attribute<-function(x,y,attributeName,attributeDefinition,measurementScale,
+                         domain,formatString,definition,unit,numberType,
```

Comment:
For each of the EML attributes (attributeName, attributeDefinition, etc.) you'll want to set the default = NULL.  This looks like `function(x,y, attributeName = NULL, attributeDefinition = NULL, ... etc.)`

The reason for setting a default is that the user doesn't have to specify these for the function to run.  A good example of this is the `data_pids = NULL` is the default in `publish_update`.  This way you don't have to actually specify data pids for it to run, and oftentimes you don't because there aren't any!

Reply:
then in your function: 
```if(!is.null(attributeName)){
eml@dataset@dataTable[[x]]@attributeList@attribute[[y]]@attributeName=attributeName
}
```